source/request-as-event-emitter.js

```diff
@@ -183,26 +180,22 @@ module.exports = (options = {}) => {
 						}, uploadEventFrequency);
 					};
 
-					// Only subscribe to `connect` event if we're actually connecting a new
-					// socket, otherwise if we're already connected (because this is a
-					// keep-alive connection) do not bother. This is important since we won't
-					// get a `connect` event for an already connected socket.
 					if (socket.connecting) {
 						socket.once('connect', onSocketConnect);
 					} else {
+						// The socket is being reused from pool,
+						// so the connect event will not be emitted
 						onSocketConnect();
 					}
-				}
+				});
 			});
 
 			if (options.gotTimeout) {
 				clearInterval(progressInterval);
 				timedOut(req, options.gotTimeout);
 			}
 
-			setImmediate(() => {
```

Comment:
Note: this setImmediate deferral was introduced in https://github.com/jstewmon/got/commit/a1eb3f7 in exchange for `get` being deferred. `get` is currently deferred (didn't track down the commit), so this does not need to be deferred.

Either `get` or `emitter.emit('request', req)` needs to be deferred to give callers a chance to subscribe to the `request` event.